### PR TITLE
Allow more field types to act as KEYs or AUTO-GEN KEYs

### PR DESCRIPTION
**Pull request by noordawod**
I added support for both of these:

ByteArray (as a KEY)
BigInteger (as AUTO-GEN KEY)

Both are perfectly capable of acting as such.

### CODE REVIEW

**Comment by noordawod** [comment]
If you think this might break long-running apps out there, I can submit a new pull and introduce two new field types (say: ByteArrayKey, BigIntegerKey) in order to be backward-compatible.

File: src/main/java/com/j256/ormlite/field/types/BigIntegerType.java [hunk 1]
```diff
@@ -66,12 +66,33 @@ public Object javaToSqlArg(FieldType fieldType, Object obj) {
 	}
 
 	@Override
+	public Object moveToNextValue(Object currentValue) {
+		BigInteger oneBigInteger = new BigInteger("1");
```
Comment (fkaelberer) [inline src/main/java/com/j256/ormlite/field/types/BigIntegerType.java]: You could use BigInteger.ONE instead.

Comment (noordawod) [inline src/main/java/com/j256/ormlite/field/types/BigIntegerType.java]: Ahh, thanks for the catch. I'll correct it right away.

Comment (noordawod) [inline src/main/java/com/j256/ormlite/field/types/BigIntegerType.java]: Should be fixed now.

**Comment by noordawod** [comment]
@j256 

Any chance you'll get around to checking this pull out?

*Note: as I mentioned before, I can also author separate field types to provide the same functionalities.*

**Comment by j256** [comment]
Merged the BigInteger stuff by hand.  Added some tests and had to tweak it a bit.  Thanks. See: https://github.com/j256/ormlite-core/commit/5e76ba72f30ac7c82513c87da25d8c9a18d6741b

**Comment by j256** [comment]
For now I'm not going to accept the byte array stuff.  Somehow it crosses the line.  Thanks.

**Comment by noordawod** [comment]
Thanks @j256 :)

**Comment by j256** [comment]
FYI: Actually I did allow `byte[]` array to be id.

**Pull request closed without merging**